### PR TITLE
chore: hello Java8

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This PR is the result of running Java8 migrations via IntelliJ (with some occasional fine-tuning).

### CODE REVIEW

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkCommitBatch.java [hunk 1]
```diff
@@ -127,20 +122,17 @@ void setMaxBatchSize(int size) {
...
+          com.google.rpc.Status.Builder status =
+              com.google.rpc.Status.newBuilder()
+                  .setCode(exception.getStatusCode().getCode().ordinal())
+                  .setMessage(exception.getMessage());
+          BatchWriteResponse.Builder response1 = BatchWriteResponse.newBuilder();
+          for (int i = 0; i < pendingOperations.size(); ++i) {
+            response1.addWriteResults(com.google.firestore.v1.WriteResult.getDefaultInstance());
+            response1.addStatus(status);
           }
+          return response1.build();
```
Comment (kolea2) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkCommitBatch.java:135]: nit - would rename this to something more descriptive, ie `batchWriteResponse`

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java [hunk 2]
```diff
@@ -305,7 +305,7 @@ static Object convertToPlainJavaTypes(Object object) {
       try {
         result =
             (rawType == Map.class)
-                ? new HashMap<String, Object>()
+                ? new HashMap<>()
```
Comment (kolea2) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java:308]: I know it's not needed, but do you think we could gain any readability here from being explicit?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java [hunk 3]
```diff
@@ -185,11 +185,6 @@ ResourcePath createPathWithSegments(ImmutableList<String> segments) {
   }
 
   public static Comparator<ResourcePath> comparator() {
-    return new Comparator<ResourcePath>() {
-      @Override
-      public int compare(ResourcePath left, ResourcePath right) {
-        return left.compareTo(right);
-      }
-    };
+    return Comparator.naturalOrder();
```
Comment (kolea2) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java:188]: just want to confirm this is tested so there's no surprises

**Review by kolea2 — APPROVED** [review]
LGTM with a few questions/nits

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java (hunk 2 above)
Comment (kolea2) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java:308]: nvm, the code above was hidden. This is fine 👍

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java (hunk 3 above)
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java:188]: The code is equivalent as it ends up calling `compareTo`. The tests for this are in https://github.com/googleapis/java-firestore/blob/c5e43aacb1c8977b475b9f3eac9be6ab79c31975/google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java

**Review by schmidt-sebastian — COMMENTED** [review]
Thanks for the review!

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java (hunk 3 above)
Comment (kolea2) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/ResourcePath.java:188]: sgtm!

**Pull request merged**